qiskit/qpy/binary_io/circuits.py

```diff
@@ -236,6 +243,19 @@ def _read_instruction(file_obj, circuit, registers, custom_operations, version,
         )
         params.append(param)
 
+    # Load Duration and Unit
+    if version >= 10:
+        # Load Duration
+        duration = instruction.duration
+        if duration == 0:
+            duration = None
+        # Load Unit
+        unit = file_obj.read(instruction.unit_size).decode(common.ENCODE)
```

Comment:
Sorry for the delay this fell through the cracks at the end of last year and I missed replying here until now. What I meant is that right now you've got an arbitrary width encoding for the unit field. But instead of doing this you can do a fixed single character encoding of the unit in the actual QPY payload. So you would replace `unit_size` and the string decoding by putting a single character at the end of the struct packing. This will be the most efficient representation because it's just a single byte and then it's just about mapping that to the units that go in the instruction object's unit field.